### PR TITLE
Cleanup UTF7Encoding negative tests

### PR DESCRIPTION
**Pull request by hughbe**
- Make EncodingHelpers.cs more extensible

/cc @tarekgh @stephentoub

### CODE REVIEW

File: src/System.Text.Encoding/tests/EncodingTestHelpers.cs [hunk 1]
```diff
@@ -10,6 +11,12 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
+        public static IEnumerable<object[]> Encodings_TestData()
+        {
+            yield return new object[] { new UnicodeEncoding() };
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: do we need to test Unicode encoding here?

**Comment by tarekgh** [comment]
LGTM

**Comment by tarekgh** [comment]
test Innerloop OSX Debug Build and Test please

**Comment by tarekgh** [comment]
Test Innerloop OSX Debug Build and Test please

File: src/System.Text.Encoding/tests/EncodingTestHelpers.cs [hunk 2]
```diff
@@ -10,6 +11,12 @@ public static class EncodingHelpers
     {
         private static readonly RandomDataGenerator s_randomGenerator = new RandomDataGenerator();
 
+        public static IEnumerable<object[]> Encodings_TestData()
```
Comment (stephentoub) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: With this change, you're converting EncodingHelpers into actually having tests.  Is the plan to remove the per-Encoding tests (except for special ones) and augment this TestData with an entry for each encoding?  Should EncodingHelpers be renamed then?

Comment (hughbe) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: The plan is only to have the negative (Invalid) tests in EncodingHelpers.cs
Positive tests will use the helper methods to ensure full coverage of all methods, but not actually have test data for positive tests

Comment (hughbe) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: So in essence, the file contains:
- Invalid tests: have test data and test code
- Valid positive tests: have only test code, not data

Maybe we split the file up then?

Comment (stephentoub) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: Yes, then we should have an NegativeEncodingTests.cs or something like that which has the negative tests for all of the encodings and nothing related to positive tests, and then the individual UTFWhateverEncodingTests.cs files would each contain the positive tests for that specific encoding.

Comment (hughbe) [inline src/System.Text.Encoding/tests/EncodingTestHelpers.cs]: Fixed in latest commit (rebased as well)

**Comment by stephentoub** [comment]
LGTM. Thanks.

**Pull request merged**